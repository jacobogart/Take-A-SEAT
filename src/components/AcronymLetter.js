import React, { Component } from 'react';
import AcronymInfo from './AcronymInfo';

class AcronymLetter extends Component {
  constructor() {
    super() 
    this.state = {
      showDetails: false
    }
  }

  toggleShowDetails = () => {
    this.setState({ 
      showDetails: !this.state.showDetails 
    })
  }

  render() {
    let { letter, name, description, starterWords } = this.props.stage;
    return (
      <div
        className="letter"
        onMouseOver={this.toggleShowDetails}
        onMouseLeave={this.toggleShowDetails}
        onClick={() => this.props.findNextWords(starterWords)}
      >
        <h2>{letter}</h2>
        {this.state.showDetails && (
          <AcronymInfo stageName={name} stageDescription={description} />
        )}
      </div>
    );
  }
}

export default AcronymLetter;
