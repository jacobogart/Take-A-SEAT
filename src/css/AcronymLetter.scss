.letter {
  display: inline-block;
  font-size: 100px;
  margin: 10px;
  position: relative;
  h2 {
    margin: 30px 0 -20px;
    font-family: $header-font;
    font-weight: normal;
    &:hover {
      cursor: pointer;
    }
  }
}

