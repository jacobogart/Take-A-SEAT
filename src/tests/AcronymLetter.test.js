import React from "react";
import AcronymLetter from "../components/AcronymLetter";
import { shallow } from "enzyme";

const mockStage = {
  "name": "Setup",
  "letter": "S",
  "description": "Setup the conditions required to execute the action on your Subject Under Test",
  "starterWords": [
    "describe",
    "let",
    "beforeEach",
    "it",
    "const"
  ],
  "id": 1
};
const mockFindNextWords = jest.fn();

describe('AcronymInfo', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <AcronymLetter
        key={mockStage.id}
        stage={mockStage}
        findNextWords={mockFindNextWords} 
      />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render AcronymInfo when showDetails is true', () => {
    expect(wrapper.state("showDetails")).toEqual(false);
    wrapper.instance().setState({ showDetails: true});
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a default state', () => {
    expect(wrapper.state("showDetails")).toEqual(false);
  });

  it('should toggle showDetails as the mouse enters and leaves', () => {
    expect(wrapper.state("showDetails")).toEqual(false);
    wrapper.find('.letter').simulate('mouseOver');
    expect(wrapper.state("showDetails")).toEqual(true);
    wrapper.find('.letter').simulate('mouseLeave');
    expect(wrapper.state("showDetails")).toEqual(false);
  });

  it('should invoke findNextWords on click', () => {
    wrapper.find('.letter').simulate('click');
    expect(mockFindNextWords).toHaveBeenCalled();
    expect(mockFindNextWords).toHaveBeenCalledWith([
      "describe",
      "let",
      "beforeEach",
      "it",
      "const"
    ]);
  });

  it('should toggle showDetails when toggleShowDetails is invoked', () => {
    expect(wrapper.state("showDetails")).toEqual(false);
    wrapper.instance().toggleShowDetails();
    expect(wrapper.state("showDetails")).toEqual(true);
    wrapper.instance().toggleShowDetails();
    expect(wrapper.state("showDetails")).toEqual(false);

  })
})