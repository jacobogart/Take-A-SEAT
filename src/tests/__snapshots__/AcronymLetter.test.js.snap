// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcronymInfo should match snapshot 1`] = `
<div
  className="letter"
  onClick={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <h2>
    S
  </h2>
</div>
`;

exports[`AcronymInfo should render AcronymInfo when showDetails is true 1`] = `
<div
  className="letter"
  onClick={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <h2>
    S
  </h2>
  <AcronymInfo
    stageDescription="Setup the conditions required to execute the action on your Subject Under Test"
    stageName="Setup"
  />
</div>
`;
